core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -408,7 +408,9 @@ static TEE_Result tee_ta_load_check_shdr(struct shdr *shdr)
 				SHDR_GET_SIG(shdr), shdr->sig_size);
 out:
 	crypto_ops.acipher.free_rsa_public_key(&key);
-	return res;
+	if (res != TEE_SUCCESS)
+		return TEE_ERROR_SECURITY;
+	return TEE_SUCCESS;
```

Comment:
Agreed, I see no security weakness at this point. Please slightly reword the commit message.
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`